File: db4oj/core/src/com/db4o/internal/activation/ActivationMode.java
package com.db4o.internal.activation;

public final class ActivationMode {
	
	public static final ActivationMode ACTIVATE = new ActivationMode();
	
	public static final ActivationMode DEACTIVATE = new ActivationMode();
	
	public static final ActivationMode PEEK = new ActivationMode();

	public static final ActivationMode PREFETCH = new ActivationMode();
	
	public static final ActivationMode REFRESH = new ActivationMode();
	
	private ActivationMode() {
	}
	
	public String toString() {
		if (isActivate()) {
			return "ACTIVATE";
		}
		if (isDeactivate()) {
			return "DEACTIVATE";
		}
		if (isPrefetch()) {
			return "PREFETCH";
		}
		if (isRefresh()) {
			return "REFRESH";
		}
		return "PEEK";
	}

	public boolean isDeactivate() {
		return this == DEACTIVATE;
	}

	public boolean isActivate() {
		return this == ACTIVATE;
	}

	public boolean isPeek() {
		return this == PEEK;
	}

	public boolean isPrefetch() {
		return this == PREFETCH;
	}
	
	public boolean isRefresh() {
		return this == REFRESH;
	}
}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Activation/ActivationMode.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

namespace Db4objects.Db4o.Internal.Activation
{
	public sealed class ActivationMode
	{
		public static readonly Db4objects.Db4o.Internal.Activation.ActivationMode Activate
			 = new Db4objects.Db4o.Internal.Activation.ActivationMode();

		public static readonly Db4objects.Db4o.Internal.Activation.ActivationMode Deactivate
			 = new Db4objects.Db4o.Internal.Activation.ActivationMode();

		public static readonly Db4objects.Db4o.Internal.Activation.ActivationMode Peek = 
			new Db4objects.Db4o.Internal.Activation.ActivationMode();

		public static readonly Db4objects.Db4o.Internal.Activation.ActivationMode Prefetch
			 = new Db4objects.Db4o.Internal.Activation.ActivationMode();

		public static readonly Db4objects.Db4o.Internal.Activation.ActivationMode Refresh
			 = new Db4objects.Db4o.Internal.Activation.ActivationMode();

		private ActivationMode()
		{
		}

		public override string ToString()
		{
			if (IsActivate())
			{
				return "ACTIVATE";
			}
			if (IsDeactivate())
			{
				return "DEACTIVATE";
			}
			if (IsPrefetch())
			{
				return "PREFETCH";
			}
			if (IsRefresh())
			{
				return "REFRESH";
			}
			return "PEEK";
		}

		public bool IsDeactivate()
		{
			return this == Deactivate;
		}

		public bool IsActivate()
		{
			return this == Activate;
		}

		public bool IsPeek()
		{
			return this == Peek;
		}

		public bool IsPrefetch()
		{
			return this == Prefetch;
		}

		public bool IsRefresh()
		{
			return this == Refresh;
		}
	}
}
